src/common/paddle/pricing.ts

```diff
@@ -0,0 +1,125 @@
+import { ConnectionManager } from '../../entity';
+import { getExperimentVariant } from '../../entity/ExperimentVariant';
+import { EntityNotFoundError } from 'typeorm';
+import { CountryCode, TimePeriod } from '@paddle/paddle-node-sdk';
+import { AuthContext } from '../../Context';
+import { createHmac } from 'node:crypto';
+import { generateStorageKey, StorageKey, StorageTopic } from '../../config';
+import {
+  PricingPreview,
+  PricingPreviewLineItem,
+} from '@paddle/paddle-node-sdk/dist/types/entities/pricing-preview';
+import { getRedisObject, setRedisObjectWithExpiry } from '../../redis';
+import { paddleInstance } from './index';
+import { ONE_HOUR_IN_SECONDS } from '../constants';
+
+export const PLUS_FEATURE_KEY = 'plus_pricing_ids';
+export const DEFAULT_PLUS_METADATA = 'plus_default';
+
+export interface PlusPricingMetadata {
+  appsId: string;
+  title: string;
+  caption?: {
+    copy: string;
+    color: string;
+  };
+  idMap: {
+    paddle: string;
+    ios: string;
+  };
+}
+
+export interface PricePreview {
+  amount: number;
+  formatted: string;
+}
+
+export interface PlusPricingPreview {
+  metadata: PlusPricingMetadata;
+  productId: string;
+  price: PricePreview & { monthly: PricePreview };
+  currency: {
+    code: string;
+    symbol: string;
+  };
+  duration: string;
+  trialPeriod?: TimePeriod | null;
+}
+
+export const getPlusPricingMetadata = async (
+  con: ConnectionManager,
+  variant: string,
+): Promise<PlusPricingMetadata[]> => {
+  const experiment = await getExperimentVariant(con, PLUS_FEATURE_KEY, variant);
+
+  if (!experiment) {
+    throw new EntityNotFoundError('ExperimentVariant not found', {
+      feature: PLUS_FEATURE_KEY,
+      variant,
+    });
+  }
+
+  try {
+    return JSON.parse(experiment.value);
+  } catch (error) {
+    throw new Error('Invalid experiment JSON value');
+  }
+};
+
+export const getPlusPricePreview = async (ctx: AuthContext, ids: string[]) => {
```

Comment:
Caching was based on how it originally was made from the original query. I just added the `ids` from the parameters to be part of the redis key.